### PR TITLE
Add details about built-in shuffleboard plugins and widgets

### PR DESCRIPTION
**Pull request by carbotaniuman**

### CODE REVIEW

**Review by Daltz333 — APPROVED** [review]

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/builtin-plugins.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
+Built-in Widgets
```
Comment (sciencewhiz) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/builtin-plugins.rst:1]: Shouldn't this be Built-in Plugins?

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+example, a theoretical “Yellow” theme could be placed in
+
+::
+
+   ~/Shuffleboard/themes/Yellow/yellowtheme.css
+
+All the stylesheets in the directory will be treated as part of the
+theme.
+
+Loading themes via plugins
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:27]:
```suggestion
Loading Themes via Plugins
```

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+     @Override
+     public List<Theme> getThemes() {
+       return ImmutableList.of(myTheme);
+     }
+
+   }
+
+Modifying or extending shuffleboard’s default themes
+----------------------------------------------------
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:52]:
```suggestion
Modifying or Extending Shuffleboard’s Default Themes
----------------------------------------------------
```

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+   @import "/edu/wpi/first/shuffleboard/app/dark.css";  /* CSS for the Material Dark theme */
+   @import "/edu/wpi/first/shuffleboard/app/midnight.css";  /* CSS for the Midnight theme */
+
+Note that ``base.css`` internally imports ``material.css``, and
+``light.css``, ``dark.css``, and ``midnight.css`` all import
+``base.css``, so importing ``light.css`` will implicitly import both
+``base.css`` and ``material.css`` as well.
+
+Source code for the CSS files distributed with shuffleboard
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:76]:
```suggestion
Source Code for the CSS Files
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Case overline

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+-  _material.css: https://github.com/wpilibsuite/shuffleboard/blob/master/api/src/main/resources/edu/wpi/first/shuffleboard/api/material.css
+-  _base.css: https://github.com/wpilibsuite/shuffleboard/blob/master/api/src/main/resources/edu/wpi/first/shuffleboard/api/base.css
+-  _light.css: https://github.com/wpilibsuite/shuffleboard/blob/master/app/src/main/resources/edu/wpi/first/shuffleboard/app/light.css
+-  _dark.css: https://github.com/wpilibsuite/shuffleboard/blob/master/app/src/main/resources/edu/wpi/first/shuffleboard/app/dark.css
+-  _midnight.css: https://github.com/wpilibsuite/shuffleboard/blob/master/app/src/main/resources/edu/wpi/first/shuffleboard/app/midnight.css
+
+Material design color swatches
+""""""""""""""""""""""""""""""
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:85]:
```suggestion
Material Design Color Swatches
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Use =, -, ^, ~ per style guide

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+The ``-swatch-<100|200|300|400|500>`` variables define progressively
+darker shades of the same primary color. The light theme uses the
+default shades of blue set in ``material.css``, but the dark theme
+overrides these with shades of red. ``-swatch-<|light|dark>-gray``
+defines three levels of gray to use for various background or text
+colors.
+
+Overriding the swatch colors
+""""""""""""""""""""""""""""
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:99]:
```suggestion
Overriding the Swatch Colors
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+Replacing the blue shades in the light theme with red
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:101]: Can we rename this to something shorter?

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+   .root {
+       -swatch-100: hsb(0, 80%, 98%);
+       -swatch-200: hsb(0, 80%, 88%);
+       -swatch-300: hsb(0, 80%, 78%);
+       -swatch-400: hsb(0, 80%, 68%);
+       -swatch-500: hsb(0, 80%, 58%);
+   }
+
+Replacing the red shades in the dark theme with blue
```
Comment (Daltz333) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:116]: Little bit of a long title here. Can this get renamed to something shorter?

**Review by Daltz333 — CHANGES_REQUESTED** [review]

File: source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst [hunk 9]
```diff
@@ -113,8 +84,8 @@ Replacing the blue shades in the light theme with red
        -swatch-500: hsb(0, 80%, 58%);
    }
 
-Replacing the red shades in the dark theme with blue
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Replacnig red with blue (dark)
```
Comment (AustinShalit) [inline source/docs/software/wpilib-tools/shuffleboard/custom-widgets/custom-themes.rst:87]:
```suggestion
Replacing red with blue (dark)
```

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**